make_grammar.js

```diff
@@ -242,20 +260,27 @@ module.exports = function make_grammar(dialect) {
 
                     field('consequence', repeat($._block)),
 
-                    optional(
-                        seq(
-                            $._left_delimiter,
-                            'else',
-                            $._right_delimiter,
-                            field('alternative', repeat($._block))
-                        )
-                    ),
+                    repeat($._else_with_clause),
 
-                    $._left_delimiter,
-                    'end',
-                    $._right_delimiter
+                    optional($._else_clause),
+                    prec.right(0, $._with_actions_end)
                 ),
 
+            _else_with_clause: ($) =>
+                prec.dynamic(
+                    PREC.else_with,
+                    seq(
+                        $._left_delimiter,
+                        'else', 'with',
+                        field('condition', $._pipeline),
+                        $._right_delimiter,
+                        field('option', repeat($._block))
+                    )
+                ),
+
+            _with_actions_end: ($) =>
```

Comment:
this is a generic end, so we could just call it action_end and use it everywhere?